### PR TITLE
Advent 2018: On the tension between generic code and special cases

### PR DESCRIPTION
**Pull request by acln0**

### CODE REVIEW

**Comment by acln0** [comment]
Thanks @philpearl!

File: content/advent-2018/generic-code-vs-special-cases.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+}
+```
+
+The code for `Copy` is changing a lot, even though the _meaning_
+of the code has not changed at all. Not only that, but `Copy` now
+concerns itself with platform-specific bits, and it knows about
+operating systems, networking, and so on. It used to be nice and
+generic, but it is now a difficult to maintain mess of special cases.
+
+It seems like something has gone wrong. This `Copy` _does_ accomodate
```
Comment (baijum) [inline content/advent-2018/generic-code-vs-special-cases.md]: Typo. s/accomodate/accommodate

Comment (acln0) [reply content/advent-2018/generic-code-vs-special-cases.md]: Thanks!

File: content/advent-2018/generic-code-vs-special-cases.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+		// generic code path
+	}
+}
+```
+
+The code for `Copy` is changing a lot, even though the _meaning_
+of the code has not changed at all. Not only that, but `Copy` now
+concerns itself with platform-specific bits, and it knows about
+operating systems, networking, and so on. It used to be nice and
+generic, but it is now a difficult to maintain mess of special cases.
```
Comment (baijum) [inline content/advent-2018/generic-code-vs-special-cases.md]: I think "a bit difficult" would be better.

**Comment by dgryski** [comment]
Thanks!

**Pull request merged**